src/types/user_operation.rs

```diff
@@ -17,3 +22,161 @@ pub struct UserOperation {
     pub paymaster_and_data: Bytes,
     pub signature: Bytes,
 }
+
+impl UserOperation {
+    pub fn pack(&self) -> Vec<u8> {
+        self.clone().encode_hex().into_bytes()
+    }
+
+    pub fn pack_for_signature(&self) -> Vec<u8> {
```

Comment:
This is the rust implementation of https://github.com/eth-infinitism/account-abstraction/blob/develop/contracts/interfaces/UserOperation.sol#L57-L71, right?

Reply:
Yes. You can find more here: https://github.com/eth-infinitism/account-abstraction/blob/develop/test/UserOp.ts#L35